src/core/Group.cpp

```diff
@@ -813,6 +814,70 @@ QSet<QUuid> Group::customIconsRecursive() const
     return result;
 }
 
+QList<QString> Group::usernamesRecursive(int topN) const
+{
+    QList<QString> usernames;
+
+    // Collect all usernames and sort for easy counting
+    for (Entry* entry : entriesRecursive()) {
+        usernames.append(entry->username());
+    }
+
+    std::sort(usernames.begin(), usernames.end());
```

Comment:
This is a lot of code for simple counting. I recommend using a `QHash<QString, int>` such that you can enumerate AND count at the same time. Here is some pseudocode:

```
QHash<QString, int> usernames;
for (auto entry : entriesRecursize()) {
	usernames.insert(entry->username(), ++usernames[entry->username]);
}

// usernames now contains all database usernames and their count
// create a list of QPairs and sort the QPairs by count (you did this already)
// Scrape off the top ten items of the list (you did this already)
```

Reply:
Good idea. Thanks.